### DISCUSSION TITLE
Is rendering the same Page multiple times within the same Document supported?

### DISCUSSION THREAD

**Discussion by esafford** [wojtekmaj/react-pdf#993] [Q&A]
I'm trying to add a left-hand sidebar to my React-PDF implementation (similar to the [pdf.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)), and I'm running into what appears to be a race condition around rendering the same page multiple times within the same `<Document />`.

I found a codepen from a separate issue, and made a quick [fork](https://codesandbox.io/s/vigilant-payne-vbsij2) that I should be highlighting the issue.  I see the issue most often with page 14 - so, for example, if you page back and forth between pages 13 and 14, you'll occasionally see the first page 14 rendered (when it should be rendering 4 copies of page 14).

Obviously, I'd prefer not to load a separate `<Document />` component to handle the sidebar rendering.  Is there another recommended approach for this, or should I be filing this as an issue?

Thanks! 😄

**Comment by wojtekmaj** [comment]
Found even easier way to reproduce this: just resize window when you're on page 14. Wonder if this has to do with PDF.js worker being unable to handle rendering the same page twice at the same time...
